make/Makefile

```diff
@@ -0,0 +1,103 @@
+# Copyright 2021 The cert-manager Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+SHELL := /usr/bin/env bash
+.SHELLFLAGS := -uo pipefail -c
+
+SOURCES=$(shell find . -type f -name "*.go")
```

Comment:
Here is a tiny tiny trick: with `=`, the makefile variable SOURCES is lazy-loaded every time it is used. Since it is used in about 30 different target (as part of the dependency list), the command `find . -type f -name "*.go"` is called 30 times.

```
$ time make -f make/Makefile help
make: *** No rule to make target 'help'.  Stop.
make -f make/Makefile help  0.23s user 0.25s system 101% cpu 0.477 total
```

If we instead go with a static variable declaration where the declaration expression is evaluated a single time, every `make` becomes  slightly faster (4 times faster for an existing target):

```
$ time make -f make/Makefile help
make: *** No rule to make target 'help'.  Stop.
make -f make/Makefile help  0.06s user 0.06s system 104% cpu 0.120 total
```

https://blog.jgc.org/2016/07/lazy-gnu-make-variables.html

Reply:
Good catch, will make that change :+1: